File: branches/Medusa/Source/Fantasy.Jobs/Management/IJobController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;

namespace Fantasy.Jobs.Management
{

    
    public interface IJobController
    {
        
        void StartJob(JobMetaData job);
       
        void Resume(JobMetaData job);
        
        void Cancel(Guid id);
       
        void Suspend(Guid id);
       
        void UserPause(Guid id);



        
        int GetAvailableProcessCount();

       
        bool IsJobProccessExisted(Guid id);

        JobMetaData[] GetRunningJobs();
        
        void RegisterJobEngine(IJobEngine engine);

        void SuspendAll(bool waitForExit);
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/management/IJobController.java
package fantasy.jobs.management;

import java.util.*;

import fantasy.jobs.*;

public interface IJobController
{

	void StartJob(JobMetaData job) throws Exception;

	void Resume(JobMetaData job) throws Exception;

	void Cancel(UUID id) throws Exception;

	void Suspend(UUID id) throws Exception;

	void UserPause(UUID id) throws Exception;




	int GetAvailableProcessCount() throws Exception;


	boolean IsJobProccessExisted(UUID id) throws Exception;

	JobMetaData[] GetRunningJobs() throws Exception;

	void RegisterJobEngine(IJobEngine engine) throws Exception; 

	void SuspendAll(boolean waitForExit) throws Exception; 
}